### ISSUE TITLE
Allow get_pkgconfig_variable's define_variable to take a dict

### ISSUE THREAD

**Issue by emersion** [mesonbuild/meson#6873]
The array notation `["name", "value"]` is pretty awkward. `{"name": "value"}` would be more readable.

**Comment by xclaesse** [comment]
I agree that notation is weird, it should to be improved. In addition to a dictionary we could also support `['var1=val1', 'var2=val2']` notation, just like `default_options` in `project()` and `subproject()`.

**Comment by vinipsmaker** [comment]
Is there a way to currently pass multiple options to `pkg-config --define-variable`? I've tried `dep.get_variable(pkgconfig : 'pluginsdir', pkgconfig_define : ['prefix', get_option('prefix'), 'libdir', get_option('prefix') / get_option('libdir')])`, but meson complained.

Would the dict notation help to achieve this feature?
